### PR TITLE
Address feedback for PowerBI Certification

### PR DESCRIPTION
**Pull request by guiangumpac**
### Description
Feedback from Microsoft:
image

### Issues Resolved
https://bitquill.atlassian.net/browse/AOS-272
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — APPROVED** [review]

File: .github/workflows/bi-connectors.yml [hunk 1]
```diff
@@ -11,40 +11,41 @@ jobs:
...
+    - uses: actions/checkout@v2
     - name: Pack Tableau JDBC connector
       id: pack-tableau-jdbc
       run: |
         zip -r opensearch_sql_jdbc.taco . -x *.taco
       working-directory: bi-connectors/TableauConnector/opensearch_sql_jdbc
     - name: Prepare Power BI ODBC connector
       run: |
-        cp OpenSearch.pq OpenSearch.m
+        cp OpenSearchProject-ODBC.pq OpenSearchProject-ODBC.m
```
Comment (raymond-lum) [inline .github/workflows/bi-connectors.yml:22]: Based on the commit description. This should be just "OpenSearchProject" excluding the -ODBC.

File: .github/workflows/bi-connectors.yml [hunk 2]
```diff
@@ -11,40 +11,41 @@ jobs:
...
       working-directory: bi-connectors/PowerBIConnector
     - name: Pack Power BI ODBC connector
       id: pack-powerbi-odbc-os-proj
       run: |
-        zip OpenSearchProject.mez *.png *.m *.resx *.pqm
+        zip OpenSearchProject-ODBC.mez *.png *.m *.resx *.pqm
```
Comment (raymond-lum) [inline .github/workflows/bi-connectors.yml:27]: Remove -ODBC as that only pertained to the protocol from Matt's email.

File: .github/workflows/bi-connectors.yml [hunk 3]
```diff
@@ -11,40 +11,41 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
     - name: Pack Tableau JDBC connector
       id: pack-tableau-jdbc
       run: |
         zip -r opensearch_sql_jdbc.taco . -x *.taco
       working-directory: bi-connectors/TableauConnector/opensearch_sql_jdbc
     - name: Prepare Power BI ODBC connector
```
Comment (raymond-lum) [inline .github/workflows/bi-connectors.yml:20]: Should this line specifically say for "OpenSearch Project" to match line 29 below?

File: .github/workflows/bi-connectors.yml [hunk 4]
```diff
@@ -11,40 +11,41 @@ jobs:
...
-        sed -i 's/\[DataSource.Kind="OpenSearch", Publish="OpenSearch.Publish"\]/[DataSource.Kind="AmazonOpenSearchConnector", Publish="AmazonOpenSearchConnector.Publish"]/g' OpenSearch.m
-        sed -i 's/OpenSearch.Contents/AmazonOpenSearchConnector.Contents/g' OpenSearch.m
-        sed -i 's/OpenSearch.Publish/AmazonOpenSearchConnector.Publish/g' OpenSearch.m
-        sed -i 's/OpenSearch.Icons/AmazonOpenSearchConnector.Icons/g' OpenSearch.m
-        sed -i 's/OpenSearchImpl/AmazonOpenSearchConnectorImpl/g' OpenSearch.m
-        sed -i 's/OpenSearchType/AmazonOpenSearchConnectorType/g' OpenSearch.m
-        sed -i 's/OpenSearch = \[/AmazonOpenSearchConnector = [/' OpenSearch.m
-#        sed -i 's/"opensearch-sql"/"Amazon-OpenSearch-Service"/g' OpenSearch.m
+        sed -i 's/Documentation.Name = "OpenSearch Project"/Documentation.Name = "Amazon OpenSearch Service"/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/section #"OpenSearchProject-ODBC";/section #"AmazonOpenSearchService-ODBC";/' AmazonOpenSearchService-ODBC.m
```
Comment (raymond-lum) [inline .github/workflows/bi-connectors.yml:35]: Remove -ODBC except for the protocol within the connector.

Comment (raymond-lum) [reply .github/workflows/bi-connectors.yml:35]: So OpenSearchProject and AmazonOpenSearchService. The main point was alignment to the Kind name and the Connector name inside the mez.

File: .github/workflows/bi-connectors.yml [hunk 5]
```diff
@@ -11,40 +11,41 @@ jobs:
...
+        sed -i 's/OpenSearchProject-ODBCType/AmazonOpenSearchService-ODBCType/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/#"OpenSearchProject-ODBC" = \[/#"AmazonOpenSearchService-ODBC" = [/' AmazonOpenSearchService-ODBC.m
+        sed -i 's/"opensearchproject-odbc"/"amazonopensearchservice-odbc"/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/(dsr) => "OpenSearch Project SQL ODBC"/(dsr) => "Amazon OpenSearch Service SQL ODBC"/g' AmazonOpenSearchService-ODBC.m
       working-directory: bi-connectors/PowerBIConnector
     - name: Pack Power BI ODBC connector for 'Amazon OpenSearch Service'
       id: pack-powerbi-odbc-amz-os-svc
       run: |
-        zip AmazonOpenSearchService.mez *.png *.m *.resx *.pqm
+        zip AmazonOpenSearchService-ODBC.mez *.png *.m *.resx *.pqm
```
Comment (raymond-lum) [inline .github/workflows/bi-connectors.yml:48]: Original filename was fine. It was the pq file content that needed to change.

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj [hunk 6]
```diff
@@ -36,7 +36,7 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="OpenSearch.pq">
+    <Compile Include="OpenSearchProject-ODBC.pq">
```
Comment (raymond-lum) [inline bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj:39]: Remove -ODBC

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj [hunk 7]
```diff
@@ -72,7 +72,7 @@
     <Compile Include="OdbcConstants.pqm">
       <SubType>Content</SubType>
     </Compile>
-    <Content Include="OpenSearch.query.pq">
+    <Content Include="OpenSearchProject-ODBC.query.pq">
```
Comment (raymond-lum) [inline bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj:75]: Remove -ODBC

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq [hunk 8]
```diff
@@ -1,18 +1,18 @@
 // This file contains Data Connector logic
 [Version = "1.0.0"]
-section OpenSearch;
+section #"OpenSearchProject-ODBC";
```
Comment (raymond-lum) [inline bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq:3]: Remove -ODBC

Comment (raymond-lum) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq:3]: This should be "OpenSearchProject" not just "OpenSearch"

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
 
 // When set to true, additional trace information will be written out to the User log. 
 // This should be set to false before release. Tracing is done through a call to 
 // Diagnostics.LogValue(). When EnableTraceOutput is set to false, the call becomes a 
 // no-op and simply returns the original value.
 EnableTraceOutput = false;
 
-[DataSource.Kind="OpenSearch", Publish="OpenSearch.Publish"]
-shared OpenSearch.Contents = Value.ReplaceType(OpenSearchImpl, OpenSearchType);
+[DataSource.Kind="OpenSearchProject-ODBC", Publish="OpenSearchProject-ODBC.Publish"]
```
Comment (raymond-lum) [inline bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq:11]: Please repeat the removal of -ODBC except for the protocol.

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq [hunk 10]
```diff
@@ -205,14 +205,14 @@ OpenSearch = [
...
+      #"opensearchproject-odbc" = [
+        GetDSR = (server, schema, object, optional options) => [ protocol = "opensearchproject-odbc", address = [ server = server, port = schema, useSSL = object ] ],
         GetFormula = (dsr, optional options) => () =>
           let
-            db = OpenSearch.Contents(dsr[address][server], dsr[address][port], dsr[address][useSSL])
+            db = #"OpenSearchProject-ODBC.Contents"(dsr[address][server], dsr[address][port], dsr[address][useSSL])
           in
             db,
-        GetFriendlyName = (dsr) => "OpenSearch SQL ODBC"
+        GetFriendlyName = (dsr) => "OpenSearch Project SQL ODBC"
```
Comment (raymond-lum) [inline bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq:215]: Why do we have SQL? Suggest just OpenSearch Project

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq [hunk 11]
```diff
@@ -149,7 +149,7 @@ OpenSearchImpl = (Server as text, Port as number, UseSSL as logical) as table =>
             CredentialConnectionString = EncryptedConnectionString
         ],
 
-        FullConnectionString = Diagnostics.LogValue("Connection String", ConnectionString & CredentialConnectionString & EncryptedConnectionString),
+        FullConnectionString = ConnectionString & CredentialConnectionString & EncryptedConnectionString,
```
Comment (raymond-lum) [inline bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq:152]: Is there an extra comma or is that correct M? I don't know just asking.

Comment (raymond-lum) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq:152]: With this change have we confirmed that connections still work?

File: .github/workflows/bi-connectors.yml [hunk 12]
```diff
@@ -11,40 +11,41 @@ jobs:
...
+        sed -i 's/\[DataSource.Kind="OpenSearchProject-ODBC", Publish="OpenSearchProject-ODBC.Publish"\]/[DataSource.Kind="AmazonOpenSearchService-ODBC", Publish="AmazonOpenSearchService-ODBC.Publish"]/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/OpenSearchProject-ODBC.Contents/AmazonOpenSearchService-ODBC.Contents/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/OpenSearchProject-ODBC.Publish/AmazonOpenSearchService-ODBC.Publish/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/OpenSearchProject-ODBCImpl/AmazonOpenSearchService-ODBCImpl/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/OpenSearchProject-ODBCType/AmazonOpenSearchService-ODBCType/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/#"OpenSearchProject-ODBC" = \[/#"AmazonOpenSearchService-ODBC" = [/' AmazonOpenSearchService-ODBC.m
+        sed -i 's/"opensearchproject-odbc"/"amazonopensearchservice-odbc"/g' AmazonOpenSearchService-ODBC.m
+        sed -i 's/(dsr) => "OpenSearch Project SQL ODBC"/(dsr) => "Amazon OpenSearch Service SQL ODBC"/g' AmazonOpenSearchService-ODBC.m
```
Comment (raymond-lum) [inline .github/workflows/bi-connectors.yml:43]: Would just call it "Amazon OpenSearch Service".

Comment (raymond-lum) [reply .github/workflows/bi-connectors.yml:43]: And "OpenSearch Project"

Comment (guiangumpac) [reply .github/workflows/bi-connectors.yml]: Updated in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq (hunk 11 above)
Comment (guiangumpac) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq]: There is no extra comma. I have added round brackets around the value to improve readability in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008.
The connection still works with these changes.

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq (hunk 9 above)
Comment (guiangumpac) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq (hunk 8 above)
Comment (guiangumpac) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj (hunk 7 above)
Comment (guiangumpac) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj]: > Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj (hunk 6 above)
Comment (guiangumpac) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.mproj]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: .github/workflows/bi-connectors.yml (hunk 5 above)
Comment (guiangumpac) [reply .github/workflows/bi-connectors.yml]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: .github/workflows/bi-connectors.yml (hunk 4 above)
Comment (guiangumpac) [reply .github/workflows/bi-connectors.yml]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: .github/workflows/bi-connectors.yml (hunk 3 above)
Comment (guiangumpac) [reply .github/workflows/bi-connectors.yml]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: .github/workflows/bi-connectors.yml (hunk 2 above)
Comment (guiangumpac) [reply .github/workflows/bi-connectors.yml]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: .github/workflows/bi-connectors.yml (hunk 1 above)
Comment (guiangumpac) [reply .github/workflows/bi-connectors.yml]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

File: bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq (hunk 10 above)
Comment (guiangumpac) [reply bi-connectors/PowerBIConnector/OpenSearchProject-ODBC.pq]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/27/commits/f27402a988e667b46f62712520cd5337128a3008

**Review by kylepbit — APPROVED** [review]

**Review by raymond-lum — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**